src/include/index/index_group.h

```diff
@@ -166,11 +166,12 @@ class base_index_group {
    */
   void init_for_open(const tiledb::Config& cfg) {
     tiledb::VFS vfs(cached_ctx_);
-    if (!vfs.is_dir(group_uri_)) {
+    if (!vfs.is_dir(group_uri_.string())) {
```

Comment:
Just an aside -- TileDB core should handle `std::filesystem::path` transparently.